packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMainUtils.ts

```diff
@@ -80,22 +80,17 @@ export function useIsSwitchingL2Chain() {
   const { isDepositMode } = app
 
   const { l2, isConnectedToArbitrum } = useNetworksAndSigners()
-
-  const { search } = useLocation()
-  const searchParams = new URLSearchParams(search)
-  const l2ChainIdSearchParam = searchParams.get('l2ChainId')
+  const { l2ChainId: l2ChainIdSearchParam } = useArbQueryParams()
 
   return useMemo(() => {
     if (isConnectedToArbitrum || !isDepositMode) {
       return false
     }
 
-    const parsedL2ChainId = parseInt(l2ChainIdSearchParam || '')
-
-    if (isNaN(parsedL2ChainId)) {
+    if (isNaN(l2ChainIdSearchParam || NaN)) {
```

Comment:
Let's make the check or undefined explicit

Reply:
Done